src/components/dialog/dialog.js

```diff
@@ -891,6 +891,11 @@ function MdDialogProvider($$interimElementProvider) {
      * Inject ARIA-specific attributes appropriate for Dialogs
      */
     function configureAria(element, options) {
+      // If the element is undefined, then there was no <md-dialog> element present.
+      // This can be caused by using the options.skipCompile and options.element properties.
+      // In this case, the template processing and compilation will be skipped and wont
+      // add the <md-dialog> automatically.
+      if (!element.length) return;
```

Comment:
I feel like this would be clearer if it specifically checked for the options and returned instead of checking element length and adding a comment about how it could get into this state.